NuclearData/ceNeutronData/aceDatabase/aceNeutronDatabase_class.f90

```diff
@@ -328,22 +336,26 @@ subroutine updateMajorantXS(self, E, rand)
     class(aceNeutronDatabase), intent(in) :: self
     real(defReal), intent(in)             :: E
     class(RNG), intent(inout)             :: rand
-    integer(shortInt)                     :: i, matIdx
+    integer(shortInt)                     :: idx
+    real(defReal)                         :: f
+    character(100), parameter :: Here = 'updateMajorantXS (aceNeutronDatabase_class.f90)'
 
     associate (maj => cache_majorantCache(1) )
       maj % E  = E
-      maj % xs = ZERO
 
-      do i = 1, size(self % activeMat)
-        matIdx = self % activeMat(i)
+      idx = binarySearch(self % eGrid, E)
 
-        ! Update if needed
-        if( cache_materialCache(matIdx) % E_tot /= E) then
-          call self % updateTotalMatXS(E, matIdx, rand)
-        end if
+      if(idx <= 0) then
+        call fatalError(Here,'Failed to find energy: '//numToChar(E)//&
+                             ' in unionised majorant grid')
+      end if
+
+      associate(E_top => self % eGrid(idx + 1), E_low  => self % eGrid(idx))
+        f = (E - E_low) / (E_top - E_low)
+      end associate
+
+      maj % xs = self % majorant(idx+1) * f + (ONE - f) * self % majorant(idx)
```

Comment:
I have a fear that if it happens that the `initMajorant` is not called by the user, but for some reason a call to `updateMajorantXS` is made, then the code will segment on the binary search. 

Similarly in the case a following calls are made: 
- activate with  materials 1,2,3 
- init majorant 
- 2nd activate (reactivate) with materials 1,2,3,4 
Then it is possible that the majorant will find itself in an invalid state (not representing the expected majorant).  
What we do with this will depend on what we finally decide on having `initMajorant` as a separate procedure in the interface, but it may involve some `allocated(self % eGrid)` checks to give more descriptive error message)